modules/coach/src/main/CoachPager.scala

```diff
@@ -71,7 +69,18 @@ final class CoachPager(
                 user  <- doc.getAsOpt[User]("coach")
               } yield Coach.WithUser(coach, user)
             }
-      },
+      }
+
+      case None => new Adapter[Coach](
+        collection = coll,
+        selector = selector,
+        projection = none,
+        sort = order.predicate
+      ) mapFutureList withUsers
```

Comment:
very nice dual adapter, and I like that the first one gets the user documents directly from the aggregated documents